### PR TITLE
v5.1.2 - Test API routes

### PR DESCRIPTION
**Pull request by elenalmg**
Adding files to test the API routes.

### CODE REVIEW

**Comment by elenalmg** [comment]
Hello, I need some helps with POST requests. Whenever I attempt to test one, the post function doesn't seem to receive my payload as an argument (but it receives the setup parameters as kwargs...). Here are some of the errors I'm encountering:  
image
I don't understand where it comes from because the **post** function function should be able to receive the payload...

**Comment by elenalmg** [comment]
_Note_ : Tests are not passing during the action because there is no connection to API. However, when you run test on the docker container with the command `docker exec -it betterave-backend-1 python -m pytest`, each test will pass : 
image

**Comment by thomktz** [comment]
Nice, maybe try to remove these tests from the Github Actions then @elenalmg

**Review by adlemoreau — APPROVED** [review]

File: betterave-backend/betterave_backend/app/api/users/routes.py
Comment (emanieluu) [inline betterave-backend/betterave_backend/app/api/users/routes.py:1]: Type hints

File: betterave-backend/tests/test_asso_operations.py
Comment (emanieluu) [inline betterave-backend/tests/test_asso_operations.py:1]: Type hints

File: betterave-backend/tests/test_auth.py
Comment (emanieluu) [inline betterave-backend/tests/test_auth.py:1]: type hints

File: betterave-backend/tests/test_class.py
Comment (emanieluu) [inline betterave-backend/tests/test_class.py:1]: type hints

File: betterave-backend/tests/test_class_groups.py
Comment (emanieluu) [inline betterave-backend/tests/test_class_groups.py:1]: type hints

**Review by emanieluu — COMMENTED** [review]
Don't forget type hints

**Review by cdechaux — APPROVED** [review]

**Pull request merged**